src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -192,32 +234,38 @@ object LDPrune {
     r
   }
 
-  def computeR2(x: BitPackedVector, y: BitPackedVector): Double = {
+  def computeR2(x: BitPackedVectorView, y: BitPackedVectorView): Double = {
     val r = computeR(x, y)
     val r2 = r * r
     assert(D_>=(r2, 0d) && D_<=(r2, 1d), s"R2 must lie in [0,1]. Found $r2.")
     r2
   }
 
-  private def pruneLocal(inputRDD: OrderedRDD[Locus, Variant, BitPackedVector],
-    r2Threshold: Double, windowSize: Int, queueSize: Option[Int]): OrderedRDD[Locus, Variant, BitPackedVector] = {
-    inputRDD.rdd.mapPartitions({ it =>
+  private def pruneLocal(inputRDD: OrderedRDD2, r2Threshold: Double, windowSize: Int, queueSize: Option[Int]): OrderedRDD2 = {
+    val localRowType = inputRDD.typ.rowType
+
+    inputRDD.mapPartitionsPreservesPartitioning({ it =>
       val queue = queueSize match {
-        case Some(qs) => new util.ArrayDeque[(Variant, BitPackedVector)](qs)
-        case None => new util.ArrayDeque[(Variant, BitPackedVector)]
+        case Some(qs) => new util.ArrayDeque[RegionValue](qs)
+        case None => new util.ArrayDeque[RegionValue]
       }
 
-      it.filter { case (v, bpv) =>
+      val bpvv = new BitPackedVectorView(localRowType, new RegionValueVariant(localRowType.fieldType(1).asInstanceOf[TVariant]))
```

Comment:
Allocating the RegionValueVariant inside the BitPackedVectorView will make this somewhat cleaner.